openapi/v2/core-metadata.yaml

```diff
@@ -148,12 +146,83 @@ components:
           type: object
           additionalProperties:
             $ref: '#/components/schemas/ProtocolProperties'
+    UpdateDevice:
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+        name:
+          type: string
+        description:
+          type: string
+        adminState:
+          type: string
+        operatingState:
+          type: string
+        lastConnected:
+          type: integer
+        lastReported:
+          type: integer
+        labels:
+          type: array
+          items:
+            type: string
+        location:
+          type: object
+        serviceName:
+          type: string
+        profileName:
+          type: string
+        autoEvents:
+          type: array
+          items:
+            $ref: '#/components/schemas/AutoEvent'
+        protocols:
+          type: object
+          additionalProperties:
+            $ref: '#/components/schemas/ProtocolProperties'
+        notify:
+          type: boolean
     DeviceProfile:
       description: "A profile defining a class of device to be onboarded, including its capabilities and data format."
       type: object
       properties:
-        apiVersion:
+        id:
           type: string
+          format: uuid
+        name:
+          type: string
+        description:
+          type: string
+        created:
+          type: integer
+        modified:
+          type: integer
+        manufacturer:
+          type: string
+        model:
+          type: string
+        labels:
+          type: array
+          items:
+            type: string
+        deviceResources:
+          type: array
+          items:
+            $ref: '#/components/schemas/DeviceResource'
+        deviceCommands:
+          type: array
+          items:
+            $ref: '#/components/schemas/ProfileResource'
+        coreCommands:
+          type: array
+          items:
+            $ref: '#/components/schemas/Command'
```

Comment:
Do we need the required section for all objects that can be part of request (add or update)?

Reply:
Yes, it make sense, added required section like the following snippet:
```
    AddDeviceRequest:
      allOf:
        - $ref: '#/components/schemas/BaseRequest'
      description: "A request to add a new device associated with a specific device service and conforming to a specific device profile."
      type: object
      properties:
        device:
          $ref: '#/components/schemas/Device'
      required:
        - device
```